modern/docker-compose.yml

```
@@ -1,43 +1,79 @@
+x-common-args: &common-args
+    CMAKE_VERSION_FULL: ${CMAKE_VERSION_FULL}
+    CMAKE_VERSION: ${CMAKE_VERSION}
+    PYTHON_VERSION: ${PYTHON_VERSION}
+    CONAN_VERSION: ${CONAN_VERSION}
+    DOCKER_USERNAME: ${DOCKER_USERNAME}
+    LIBSTDCPP_VERSION: ${LIBSTDCPP_VERSION}
+    ARTIFACTORY_REPOSITORY: ${ARTIFACTORY_REPOSITORY}
+    JENKINS_AGENT_VERSION: ${JENKINS_AGENT_VERSION}
+
+x-gcc5: &gcc5
+    GCC_VERSION: 5.5.0
+
+x-gcc6: &gcc6
+    GCC_VERSION: 6.5.0
+
+x-gcc7: &gcc7
+    GCC_VERSION: 7.5.0
+
+x-gcc8: &gcc8
+    GCC_VERSION: 8.4.0
+
+x-gcc9: &gcc9
+    GCC_VERSION: 9.3.0
+
+x-gcc10: &gcc10
+    GCC_VERSION: 10.3.0
+
+x-gcc11: &gcc11
+    GCC_VERSION: 11.1.0
+
+x-llvm9: &llvm9
+    LLVM_VERSION: 9.0.1
+
+x-llvm10: &llvm10
+    LLVM_VERSION: 10.0.1
+
+x-llvm11: &llvm11
+    LLVM_VERSION: 11.1.0
+
+x-llvm12: &llvm12
+    LLVM_VERSION: 12.0.0
+
+
 version: '3'
```

```suggestion
version: '3.4'
```

Just like CMake, move it to the top. Anchor is supported since 3.4. I remember I tried to use it in the very beginning of CDT and was not available. Now you revisited a powerful feature. Thanks!